crates/zksync/core/src/vm/env.rs

```diff
@@ -7,15 +7,50 @@ use zksync_multivm::{
 use zksync_state::interface::{ReadStorage, StoragePtr};
 use zksync_types::{
     block::{unpack_block_info, L2BlockHasher},
-    fee_model::L1PeggedBatchFeeModelInput,
+    fee_model::PubdataIndependentBatchFeeModelInput,
     StorageKey, SYSTEM_CONTEXT_ADDRESS, SYSTEM_CONTEXT_BLOCK_INFO_POSITION,
 };
 use zksync_utils::h256_to_u256;
 
+#[derive(Debug, Clone)]
+/// Values related to the era vm environment
+pub struct ZkEnv {
```

Comment:
If I am not mistaken simulation is essentially estimation. In that case should we include `gas_price_scale_factor` and `estimate_gas_scale_factor`? This is how we provide padding during estimation, see [here](https://matter-labs.github.io/zksync-era/core/latest/guides/advanced/07_fee_model.html#gas-estimation).

Reply:
It is for evm but not for era vm. For us this is more like a "preview" to run the script with the actual values (and verify the transaction goes through in a closer to real life environment) and also to show more accurate values to the user about the estimation. When [sending the transaction](https://github.com/matter-labs/foundry-zksync/blob/a4f0ebf8ea8f0725196b051b89dcf0ef7c49caed/crates/script/src/broadcast.rs#L136) we reestimate via the rpc call (which I believe uses the factors you mention) before sending. In a following issue maybe we could show the rpc estimation directly and not the simulation one as it is done with chains like Moonbeam where gas rules are a bit different than Ethereum.
Nonetheless if we were to use the simulation result for the final tx, the gas limit is bumped by `gas_estimate_multiplier` (see: https://github.com/matter-labs/foundry-zksync/blob/a4f0ebf8ea8f0725196b051b89dcf0ef7c49caed/crates/script/src/simulate.rs#L96) And gas price is estimated via rpc by alloy unless there's a hardcoded value (see: https://github.com/matter-labs/foundry-zksync/blob/a4f0ebf8ea8f0725196b051b89dcf0ef7c49caed/crates/script/src/broadcast.rs#L317). Those are probably the places to play with those values vs `ZkEnv` that is more the actual literal values fed to the vm (for example we could feed values via config and use them for gas reports)